src/main/java/uk/gov/digital/ho/hocs/workflow/domain/repositories/JsonFileReader.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.digital.ho.hocs.workflow.domain.repositories;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.workflow.application.LogEvent;
+import uk.gov.digital.ho.hocs.workflow.domain.exception.ApplicationExceptions;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+@Service
+public abstract class JsonFileReader {
+
+    protected final ObjectMapper objectMapper;
+
+    protected JsonFileReader(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    protected <T> T readValueFromFile(TypeReference<T> reference, String fileName) {
+        try (InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(
+            String.format("%s/%s.json", getFolderName(), fileName))) {
+            return objectMapper.readValue(in, reference);
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw new ApplicationExceptions.JsonFileReadException(
```

Comment:
This also supports moving into the ScreenRepository - we should have a proper Exception...